cypress/integration/app_spec.ts

```diff
@@ -1,70 +1,121 @@
-  cy.intercept('GET', 'https://freesound.org/apiv2/packs/26158/sounds/?fields=id%2Cname%2Cduration%2Cnum_downloads%2Cusername%2Cnum_ratings', {fixture:'pack-sounds.json'})
-  cy.intercept('GET', 'https://freesound.org/apiv2/sounds/462808/similar/?fields=id%2Cname%2Cduration%2Cnum_downloads%2Cusername%2Cnum_ratings', {fixture:'similar-sounds.json'})
-  cy.intercept('GET', 'https://freesound.org/data/previews/462/462808_8386274-lq.mp3', {fixture:'sound.mp3'})
-  cy.intercept('GET', 'https://freesound.org/apiv2/search/text/*', {fixture:'search-results.json'})
+  cy.intercept("GET", "https://freesound.org/apiv2/sounds/462808", {
+    fixture: "sound.json",
+  });
+  cy.intercept("GET", "/people/qfox123/packs/26158", { fixture: "sound.json" });
+  cy.intercept("GET", "https://freesound.org/apiv2/packs/26158/", {
+    fixture: "pack.json",
+  });
+  cy.intercept(
+    "GET",
+    "https://freesound.org/apiv2/packs/26158/sounds/?fields=id%2Cname%2Cduration%2Cnum_downloads%2Cusername%2Cnum_ratings",
+    { fixture: "pack-sounds.json" }
+  );
+  cy.intercept(
+    "GET",
+    "https://freesound.org/apiv2/sounds/462808/similar/?fields=id%2Cname%2Cduration%2Cnum_downloads%2Cusername%2Cnum_ratings",
+    { fixture: "similar-sounds.json" }
+  );
+  cy.intercept(
+    "GET",
+    "https://freesound.org/data/previews/462/462808_8386274-lq.mp3",
+    { fixture: "sound.mp3" }
+  );
+  cy.intercept(
+    "GET",
+    "https://freesound.org/apiv2/search/text/?fields=id%2Cname%2Cpreviews%2Cduration%2Cnum_downloads%2Cusername%2Cnum_ratings&page_size=10&query=hello",
+    {
+      fixture: "search-results.json",
+    }
+  );
+
+  cy.intercept(
+    "GET",
+    "https://freesound.org/apiv2/search/text/?fields=id%2Cname%2Cpreviews%2Cduration%2Cnum_downloads%2Cusername%2Cnum_ratings&page_size=10&page=2&query=hello",
+    {
+      fixture: "search-results-page-2.json",
+    }
+  );
 }
 
 describe("Home Page", () => {
   beforeEach(() => {
-    setupServerMocks()
-  })
+    setupServerMocks();
+  });
 
   it("should load the home page", () => {
     cy.visit("/");
-    cy
-      .get(`[data-e2e-id='SoundList']`)
-      .should('be.visible')
+    cy.get(`[data-e2e-id='SoundList']`).should("be.visible");
+  });
+});
+
+describe("Pagination element functionality", () => {
+  beforeEach(() => {
+    setupServerMocks();
+  });
+
+  it("checks if next page button shows next page of results", () => {
+    cy.visit("/search?q=hello");
+    cy.get(`[data-e2e-id="SoundList-track-name"]`)
+      .first()
+      .should("be.visible")
+      .should("have.text", "hello");
+    cy.get(`[data-e2e-id="SoundList-track-url"]`)
+      .first()
+      .should("be.visible")
+      .should("have.attr", "href", "/sound/123");
+    cy.get(`[data-e2e-id="first-page-button"]`).should("be.visible");
+    cy.get(`[data-e2e-id="next-page-button"]`).should("be.visible").click();
+    cy.wait(30000); // This is necessary to give enough time to re-render.
```

Comment:
This timeout seems excessive. If the timeout needs to be this long its likely that the test is not being mocked, which was an issue I ran into before. I can take a look at this

Reply:
Tests only seem to fail on push; they usually pass locally.

Reply:
I think the changes in af396bf are taking care of test cases not passing on push, so I'm going to resolve this conversation.